src/analysis/retail/shaman/restoration/modules/dragonflight/Tier30.tsx

```diff
@@ -0,0 +1,240 @@
+      this.swellingRainChainheal +
+      this.swellingRainHealingSurge +
+      this.swellingRainHealingWave +
+      this.rainstormHealing
+    );
+  }
+
+  onHealingRainCast(event: CastEvent) {
+    //apparently these 'stacks' for each buff don't appear in logs so we have to manually tally the number of tidewaters hits to know how potent our current buff is
+    this.current4pcStacks = getTidewatersHealingEvents(event).length;
+  }
+
+  on2pcHeal(event: HealEvent) {
+    this.tidewatersHealing += event.amount + (event.absorbed || 0);
+    this.tidewatersOverheal += event.overheal || 0;
+  }
+
+  on4pcHeal(event: HealEvent) {
+    if (!this.selectedCombatant.hasBuff(ITEMS.T30_RAINSTORM_BUFF.id)) {
+      return;
+    }
+    const healIncrease = this.current4pcStacks * TIER_4PC_RAINSTORM_PER_STACK;
+    this.rainstormHealing += calculateEffectiveHealing(event, healIncrease);
+  }
+
+  handleHealingWave(event: CastEvent) {
+    if (
+      !this.selectedCombatant.hasBuff(
+        ITEMS.T30_SWELLING_RAIN_BUFF.id,
+        event.timestamp,
+        CAST_BUFFER_MS,
+      )
+    ) {
+      return;
+    }
+    const healIncrease = this.current4pcStacks * TIER_4PC_SWELLING_RAIN_HSHW_PER_STACK;
+    const swellingRainHealingWaves = getSwellingRainHealingWaves(event);
+    debug && console.log('healing wave', swellingRainHealingWaves, healIncrease);
+    if (swellingRainHealingWaves.length > 0) {
+      this.swellingRainHealingWave += this._tallyHealingIncrease(
+        swellingRainHealingWaves,
+        healIncrease,
+      );
+      debug && console.log(this.swellingRainHealingWave);
+    }
+  }
+
+  handleHealingSurge(event: HealEvent) {
+    if (
+      !this.selectedCombatant.hasBuff(
+        ITEMS.T30_SWELLING_RAIN_BUFF.id,
+        event.timestamp,
+        CAST_BUFFER_MS,
+      )
+    ) {
+      return;
+    }
+    const healIncrease = this.current4pcStacks * TIER_4PC_SWELLING_RAIN_HSHW_PER_STACK;
+    this.swellingRainHealingSurge += calculateEffectiveHealing(event, healIncrease);
+  }
+
+  handleChainHeal(event: CastEvent) {
+    if (
+      !this.selectedCombatant.hasBuff(
+        ITEMS.T30_SWELLING_RAIN_BUFF.id,
+        event.timestamp,
+        CAST_BUFFER_MS,
+      )
+    ) {
+      return;
+    }
+    const healIncrease = this.current4pcStacks * TIER_4PC_SWELLING_RAIN_CHAINHEAL_PER_STACK;
+    const orderedChainHeal = this.chainHealNormalizer.normalizeChainHealOrder(event);
+    debug && console.log('chain heal: ', orderedChainHeal, healIncrease);
+    if (orderedChainHeal.length > 0) {
+      this.swellingRainChainheal += this._tallyHealingIncrease(orderedChainHeal, healIncrease);
+      debug && console.log(this.swellingRainChainheal);
+    }
+  }
+
+  private _tallyHealingIncrease(events: HealEvent[], healIncrease: number): number {
+    if (events.length > 0) {
+      return events.reduce(
+        (amount, event) => amount + calculateEffectiveHealing(event, healIncrease),
+        0,
+      );
+    }
+    return 0;
```

Comment:
you shouldn't need to do this check since you provide an initial value to the reduce.